demos/llamastack/Llama_Stack_NVIDIA_E2E_Flow.ipynb

```diff
@@ -0,0 +1,2924 @@
+      "Model nvidia-e2e-tutorial/test-messages-model@v1763646228 not available after 84.33940100669861 seconds.\n",
+      "Model nvidia-e2e-tutorial/test-messages-model@v1763646228 not available after 94.82941198348999 seconds.\n",
+      "Model nvidia-e2e-tutorial/test-messages-model@v1763646228 not available after 105.3087830543518 seconds.\n",
+      "Model nvidia-e2e-tutorial/test-messages-model@v1763646228 not available after 115.79959392547607 seconds.\n",
+      "Model nvidia-e2e-tutorial/test-messages-model@v1763646228 available after 126.27090287208557 seconds.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Check that the customized model has been picked up by NIM;\n",
+    "# We allow up to 5 minutes for the LoRA adapter to be loaded\n",
+    "wait_nim_loads_customized_model(model_id=customized_chat_model_dir)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 62,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "✅ Inference response: -----------\n",
+      "A) blue\n",
+      "B) purple\n",
+      "C) white\n",
+      "D) yellow\n",
+      "\n",
+      "The best answer\n"
+     ]
+    }
+   ],
+   "source": [
+    "import requests\n",
+    "\n",
+    "response = requests.post(\n",
+    "    f\"{NIM_URL}/v1/completions\",\n",
+    "    json={\n",
+    "        \"model\": customized_chat_model_dir,\n",
+    "        \"prompt\": \"Complete the sentence using one word: Roses are red, violets are \",\n",
+    "        \"temperature\": 0.7,\n",
+    "        \"top_p\": 0.9,\n",
+    "        \"max_tokens\": 20,\n",
+    "    }\n",
+    ")\n",
+    "\n",
+    "if response.status_code == 200:\n",
+    "    print(f\"✅ Inference response: {response.json()['choices'][0]['text']}\")\n",
+    "else:\n",
+    "    print(f\"❌ Error: {response.status_code} - {response.text}\")\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 63,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "assert len(response.content) > 1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Evaluate Customized Model with chat dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 65,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Creating new benchmark ID: test-eval-config-chat-1763646850\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Re-create benchmark ID with new timestamp (since LlamaStack pod restarted in cell 84)\n",
+    "benchmark_id = f\"test-eval-config-chat-{int(time())}\"\n",
+    "print(f\"Creating new benchmark ID: {benchmark_id}\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 66,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Re-define the eval config with the new benchmark ID\n",
```

Comment:
When you have time, please investigate how to adding safety guardrails via the llamastack server.